### PR TITLE
Firefox: Pass IgnoreHttpsErrorsTests

### LINKED ISSUE #106: Pass interception.spec.js tests in Firefox

### PR DESCRIPTION
**Pull request by Meir017**
related to #106

### CODE REVIEW

**Comment by Meir017** [comment]
@kblok there are still a few failing tests

**Comment by kblok** [comment]
@Meir017 locally or on CI?

**Comment by Meir017** [comment]
Locally, it's almost ready

File: src/PlaywrightSharp.Firefox/FirefoxInterceptableRequest.cs [hunk 1]
```diff
@@ -26,19 +28,69 @@ public FirefoxInterceptableRequest(FirefoxSession session, Frame frame, List<Req
...
+            }
         }
 
-        public Task FulfillAsync(ResponseData response)
+        public async Task FulfillAsync(ResponseData response)
         {
-            throw new System.NotImplementedException();
+            var responseHeaders = response.Headers?.ToDictionary(header => header.Key.ToLower(), header => header.Value)
+                ?? new Dictionary<string, string>();
+            if (!string.IsNullOrEmpty(response.ContentType))
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/FirefoxInterceptableRequest.cs:69]: let's add one empty line here.

File: src/PlaywrightSharp.Firefox/FirefoxNetworkManager.cs [hunk 2]
```diff
@@ -49,19 +49,19 @@ private void OnMessageReceived(object sender, IFirefoxEvent e)
 
         private void OnRequestWillBeSent(NetworkRequestWillBeSentFirefoxEvent e)
         {
-            _requests.TryGetValue(e.RedirectedFrom ?? string.Empty, out var redirectId);
-            var frame = (redirectId != null ? redirectId.Request.Frame : (e.FrameId != null ? _page.FrameManager.Frames[e.FrameId] : null)) as Frame;
+            _requests.TryGetValue(e.RedirectedFrom ?? string.Empty, out var redirected);
+            var frame = redirected?.Request?.Frame ?? (e.FrameId != null ? _page.FrameManager.Frames[e.FrameId] : null);
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/FirefoxNetworkManager.cs:53]: I think this is not accurate with pw.

```suggestion
            var frame = redirected != null ? redirected.Request.Frame : (e.FrameId != null ? _page.FrameManager.Frames[e.FrameId] : null);
```

**Comment by kblok** [comment]
We have many tests failing related to this feature.

**Comment by Meir017** [comment]
I'll investigate this more, the tests passed locally so I figured they are okay. I'll test again

**Comment by kblok** [comment]
Reducing the scope of this PR @Meir017 to pass only IgnoreHttpsErrorsTests. 

If seems some feature in some test is breaking other tests. We will pass one by one till we get the guilty one.

**Pull request merged**